### PR TITLE
Feature Request: Add new API to return Product Attributes with Swatches data.

### PR DESCRIPTION
**Pull request by nikunjkotecha**

### CODE REVIEW

**Comment by nikunjkotecha** [comment]
cc @sdelbosc

**Comment by sdelbosc** [comment]
To give a bit more context around this one.
We need to get swatches details along with product options but Magento does not give them and does not seem to allow us to override that to just add what we miss.
So far we have gone for a direct integration between Drupal and Magento but the Magento side shouldn't change much that's why we raised the PR. I am going to raise a proper feature request for that.

File: Model/Entity/Attribute.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * Acquia/CommerceManager/Model/Product/Attribute/OptionManagement.php
```
Comment (malachy-mcconnell) [inline Model/Entity/Attribute.php]: Path needs updated

File: Model/Entity/Attribute.php [hunk 2]
```diff
@@ -1,16 +1,15 @@
 <?php
 
 /**
- * Acquia/CommerceManager/Model/Product/Attribute/OptionManagement.php
```
Comment (sdelbosc) [inline Model/Entity/Attribute.php]: @nikunjkotecha Do we need to do this on our current customer?

Comment (nikunjkotecha) [reply Model/Entity/Attribute.php]: yes, i was hoping to finish cleanup soon and start using via composer's , will create PR anyway

Comment (malachy-mcconnell) [reply Model/Entity/Attribute.php]: I would prefer these attributes to be added using the extension_attributes. Maybe that can be a follow-up task to convert it to extension_attributes. If we merge this PR do we also need a new API endpoint in the Commerce Connector, yes? I wonder if we can use an existing endpoint and get Magento to add the swatch attributes in with the other attributes.

Please let me know your opinion. I'm prepared to merge this PR as-is if we create a follow up ticket to create a better Magento-side implementation later and a ticket for the Commerce Connector (although ideally a new endpoint is not needed)

What do you think?

**Comment by vbouchet31** [comment]
@malachy-mcconnell @sdelbosc Is there any PF ticket for this to be integrated in ACM? As of now, on our project we are using direct integration with MDC to get product option to workaround.

**Comment by sdelbosc** [comment]
@vbouchet31 I am not fully clear but given this has been merged on the ACM magento repo I would assume it is part of product already

**Comment by vbouchet31** [comment]
@sdelbosc My understanding is this swatch attribute should be added to the product option via the arbitrary extension (or whatever is the name of this array where we can put whatever we want without it being filtered by ACM). @nikunjkotecha may have a better view on it.

**Pull request merged**